packages/ng-query/src/lib/base-query.ts

```diff
@@ -0,0 +1,82 @@
+import {
+  DefaultedQueryObserverOptions,
+  notifyManager,
+  QueryKey,
+  QueryObserver,
+  QueryObserverResult,
+} from '@tanstack/query-core';
+import { Observable } from 'rxjs';
+
+export function notify<
+  TQueryFnData = unknown,
+  TError = unknown,
+  TData = TQueryFnData,
+  TQueryData = TQueryFnData,
+  TQueryKey extends QueryKey = QueryKey
+>(
+  options: DefaultedQueryObserverOptions<
+    TQueryFnData,
+    TError,
+    TData,
+    TQueryData,
+    TQueryKey
+  >
+) {
+  options.onError &&= notifyManager.batchCalls(options.onError);
+
+  options.onSuccess &&= notifyManager.batchCalls(options.onSuccess);
+
+  options.onSettled &&= notifyManager.batchCalls(options.onSettled);
+}
+
+export function baseResults<
+  TQueryFnData = unknown,
+  TError = unknown,
+  TData = TQueryFnData,
+  TQueryData = TQueryFnData,
+  TQueryKey extends QueryKey = QueryKey
+>(
+  queryObserver: QueryObserver<
+    TQueryFnData,
+    TError,
+    TData,
+    TQueryData,
+    TQueryKey
+  >,
+  queryKey: TQueryKey,
+  options: DefaultedQueryObserverOptions<
+    TQueryFnData,
+    TError,
+    TData,
+    TQueryData,
+    TQueryKey
+  >,
+  onDestroy: () => void
+): Observable<QueryObserverResult<TData, TError>> {
```

Comment:
```suggestion
): Observable<QueryObserverResult<TData, TError>> {
```

That's not correct in case of infinite observer

Reply:
I had to take out the types, there is no way i could infer `QueryObserverResult` or `InfiniteQueryObserverResult`. So I ended up leaving out the output type. Same goes for `QueryObserver` parameter.